### PR TITLE
[DUOS-2510] Redesign the match results to show Abstain

### PR DESCRIPTION
**Pull request by kaitlynoflaherty**
### Addresses
https://broadworkbench.atlassian.net/jira/software/c/projects/DUOS/boards/123?modal=detail&selectedIssue=DUOS-2510

### Summary
image
----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/utils/BucketUtils.js [hunk 1]
```diff
@@ -238,6 +258,17 @@ const calculateAlgorithmResultForBucket = (bucket) => {
   }
 };
 
+/**
+* Process the match results for V3 Abstain. If we have a V3 result and we have
+* an ABSTAIN case, we can return true if the number of abstentions > 1
```
Comment (fboulnois) [inline src/utils/BucketUtils.js]:
```suggestion
* an ABSTAIN case, we can return true if the number of abstentions > 0
```

File: src/utils/BucketUtils.js [hunk 2]
```diff
@@ -189,45 +189,65 @@ const filterDatasetsByDACs = (dacIds, datasets) => {
...
  * @private
  * @param bucket
  * @returns {{result: string, failureReasons: string[], id, createDate: undefined}|{result: (string|string), failureReasons: *, id: *, createDate: *}|{result: string, failureReasons: undefined, id, createDate: undefined}}
  */
 const calculateAlgorithmResultForBucket = (bucket) => {
-
-  // We actually DO NOT want to show system match results when the data use indicates
+  // V1 and V2: We actually DO NOT want to show system match results when the data use indicates
   // that a match should not be made. This happens for all "Other" cases.
+  const algorithmVersionV3 = bucket.matchResults.algorithmVersion === 'v3' ? true : false;
```
Comment (fboulnois) [inline src/utils/BucketUtils.js]:
```suggestion
  const algorithmVersionV3 = bucket.matchResults.algorithmVersion === 'v3';
```

File: src/utils/BucketUtils.js [hunk 3]
```diff
@@ -189,45 +189,65 @@ const filterDatasetsByDACs = (dacIds, datasets) => {
...
   const unmatchable = isOther(bucket.dataUse) || shouldAbstain(bucket.dataUse);
   // Check on all possible true/false values in the matches.
   // If all matches are the same, we can merge them into a single match object for display.
   // If they are not all the same, we have to punt this decision solely to the DAC.
-  const matchVals = unmatchable ? [] : flow(
+  // Check algorithm version: V3 does not need to be checked for 'unmatchable'
```
Comment (fboulnois) [inline src/utils/BucketUtils.js:210]: The nested ternaries make this a little hard to follow. I think this could be expressed as:

```
const matchVals = (algorithmVersionV3 || !unmatchable) ? flow(
      map(m => m.match),
      uniq
    )(bucket.matchResults)) : [];
```

**Review by fboulnois — COMMENTED** [review]
A few suggestions:

File: src/utils/BucketUtils.js (hunk 3 above)
Comment (kaitlynoflaherty) [reply src/utils/BucketUtils.js:210]: Good point! Thanks for the suggestion

**Review by rushtong — APPROVED** [review]
👍🏽

**Review by fboulnois — APPROVED** [review]
👍
